src/NServiceBus.Core/Transports/Msmq/ReceiveWithNativeTransaction.cs

```diff
@@ -20,6 +20,11 @@ class ReceiveWithNativeTransaction : ReceiveStrategy
 
                     var message = inputQueue.Receive(TimeSpan.FromMilliseconds(10), msmqTransaction);
 
+                    if (message == null)
```

Comment:
Added it mostly to make R# happy but I think we should keep it to be safe or decompile the MSMQ client and make sure that it can never return null?